de.metas.adempiere.adempiere/client/src/main/java-legacy/org/compiere/apps/AEnv.java

```diff
@@ -633,30 +633,33 @@ public static void zoom(
 		{
 			return;
 		}
-		
+
 		zoom(RecordZoomWindowFinder.newInstance(TableName, Record_ID)
 				.setSO_Window_ID(AD_Window_ID)
 				.setPO_Window_ID(PO_Window_ID));
 	}
-	
+
 	private static final void zoom(final RecordZoomWindowFinder zoomInfo)
 	{
 		final int windowIdToUse = zoomInfo.findAD_Window_ID();
-		if(windowIdToUse <= 0)
+		if (windowIdToUse <= 0)
 		{
 			log.warn("No AD_Window_ID found to zoom for {}", zoomInfo);
 			return;
 		}
-		
+
 		final MQuery query = zoomInfo.createZoomQuery();
-		
-		final AWindow frame = new AWindow();
-		if (!frame.initWindow(windowIdToUse, query))
-		{
-			return;
-		}
-		addToWindowManager(frame);
-		showCenterScreenOrMaximized(frame);
+
+		// task #797 Make sure the windows are displayed in the same thread as the process
```

Comment:
Hello.
Right. As I said yesterday, I don't understand what thread interferes with the process but we found out while debugging that a field had a certain value in an if body that was defined on that field to have a different value. So we realized it's from another thread. But I don't know why. Teo suggested to try this and see if it works. It did. 
Please, fix the comment to be more easy to understand. I didn't know how to write it right.